eNKay2408/POS-System
# test: Add Unit Test for ViewModels

### Description
- This pull request adds comprehensive unit tests for the **ViewModels** in the POSSystem project. 
- The following ViewModels have been covered in **6 classes**: RegisterViewModel, LoginViewModel, BrandViewModel, CategoryViewModel, ProductViewModel, AddProductViewModel.

### Changes
  - Added unit tests for RegisterViewModel to check password strength, email validation, and employee registration. 
  - Added unit tests for LoginViewModel to test login, Google authentication, and credential management. 
  - Added unit tests for BrandViewModel to ensure proper loading and addition of brands. 
  - Added unit tests for CategoryViewModel to verify category loading and addition functionality. 
  - Added unit tests for ProductViewModel to validate product loading, filtering, sorting, and payment processing. 
  - Added unit tests for AddProductViewModel to ensure categories and brands are loaded correctly and products are saved properly.

### How to check?
- Open Visual Studio's **Test Explorer** option
- Click the **run all tests** button to see the results
![image](https://github.com/user-attachments/assets/6a9f37ba-ba99-4bf1-850c-beea4399ea76)
